src/gui/sdl_mapper.cpp

```diff
@@ -2990,37 +2993,67 @@ void MAPPER_DisplayUI() {
 	GFX_EndUpdate( nullptr );
 	mapper.window = GFX_GetWindow();
 	if (mapper.window == nullptr) {
-		E_Exit("Could not initialize video mode for mapper: %s", SDL_GetError());
+		E_Exit("MAPPER: Could not initialize video mode: %s",
+		       SDL_GetError());
 	}
 	mapper.renderer = SDL_GetRenderer(mapper.window);
 #if C_OPENGL
 	SDL_GLContext context = nullptr;
 	if (!mapper.renderer) {
 		context         = SDL_GL_GetCurrentContext();
-		mapper.renderer = SDL_CreateRenderer(mapper.window, -1, 0);
+		if (!context) {
+			E_Exit("MAPPER: Failed to retrieve current OpenGL context: %s",
+			       SDL_GetError());
+		}
+
+		const auto renderer_drivers_count = SDL_GetNumRenderDrivers();
+		if (renderer_drivers_count < 0) {
+			E_Exit("MAPPER: Failed to retrieve available SDL renderer drivers: %s",
+			       SDL_GetError());
+		}
+		int renderer_driver_index = -1;
+		for (int i = 0; i < renderer_drivers_count; i++) {
+			SDL_RendererInfo renderer_info;
+			if (SDL_GetRenderDriverInfo(i, &renderer_info) < 0) {
+				E_Exit("MAPPER: Failed to retrieve SDL renderer driver info: %s",
+				       SDL_GetError());
+			}
+			if (strcmp(renderer_info.name, "opengl") == 0) {
+				renderer_driver_index = i;
+				break;
+			}
+		}
+		if (renderer_driver_index == -1) {
+			E_Exit("MAPPER: OpenGL support in SDL renderer is unavailable but required for OpenGL output");
+		}
+		mapper.renderer = SDL_CreateRenderer(mapper.window,
+		                                     renderer_driver_index,
+		                                     0);
 	}
 #endif
 	if (mapper.renderer == nullptr) {
-		E_Exit("Could not retrieve window renderer for mapper: %s",
+		E_Exit("MAPPER: Could not retrieve window renderer: %s",
 		       SDL_GetError());
 	}
 
-	SDL_RenderSetLogicalSize(mapper.renderer, 640, 480);
+	if (SDL_RenderSetLogicalSize(mapper.renderer, 640, 480) < 0) {
+		LOG_WARNING("MAPPER: Failed to set renderer logical size: %s",
+		            SDL_GetError());
+	}
 
 	// Create font atlas surface
 	SDL_Surface* atlas_surface = SDL_CreateRGBSurfaceFrom(
 	        int10_font_14, 8, 256 * 14, 1, 1, 0, 0, 0, 0);
 	if (atlas_surface == nullptr) {
-		E_Exit("Failed to create atlas surface for mapper: %s",
-		       SDL_GetError());
+		E_Exit("MAPPER: Failed to create atlas surface: %s", SDL_GetError());
 	}
 
 	// Invert default surface palette
 	const SDL_Color atlas_colors[2] = {{0x00, 0x00, 0x00, 0x00},
 	                                   {0xff, 0xff, 0xff, 0xff}};
 	if (SDL_SetPaletteColors(atlas_surface->format->palette, atlas_colors, 0, 2) <
 	    0) {
-		LOG_WARNING("Failed to set colors in font atlas: %s",
+		LOG_WARNING("MAPPER: Failed to set colors in font atlas: %s",
```

Comment:
nice error checking additions and messaging cleanup!

Reply:
+1, that's great stuff @GranMinigun 🎉